homeassistant/components/caldav/config_flow.py

```diff
@@ -0,0 +1,176 @@
+"""Config flow to configure caldav."""
+import logging
+
+import voluptuous as vol
+from yarl import URL as yurl
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME, CONF_VERIFY_SSL
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+
+from . import async_caldav_connect
+from .const import (
+    CALDAV_EXCEPTIONS,
+    CONF_ADD_CUSTOM_CALENDAR,
+    CONF_CALENDARS,
+    CONF_CUSTOM_CALENDARS,
+    CONF_DAYS,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        # pylint: disable=no-value-for-parameter
+        vol.Required(CONF_USERNAME): selector.TextSelector(),
+        vol.Required(CONF_PASSWORD): selector.TextSelector(
+            selector.TextSelectorConfig(type=selector.TextSelectorType.PASSWORD)
+        ),
+        vol.Required(CONF_URL): selector.TextSelector(
+            selector.TextSelectorConfig(type=selector.TextSelectorType.URL)
+        ),
+        vol.Optional(
+            CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL
+        ): selector.BooleanSelector(),
+        vol.Optional(CONF_DAYS, default=1): cv.positive_int,
+        vol.Optional(CONF_CALENDARS, default=[]): selector.SelectSelector(
+            selector.SelectSelectorConfig(options=[], custom_value=True, multiple=True),
+        ),
+        vol.Required(CONF_ADD_CUSTOM_CALENDAR, default=False): bool,
+    }
+)
+
+CUSTOM_CALENDARS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(
+            CONF_CUSTOM_CALENDARS,
```

Comment:
One thing i've done in google calendar is steer people away from all of the integration specific customizations like making custom calendars (relying on search and offsets and stuff, to try to find more general solutions that apply to each calendar integration.

(1) We can support renaming/disabling if you give the entities unique ids
(2) The search functionality can apply across all calendar types.  These binary sensors also have a bunch of limitations and gaps e.g. missing data for concurrent events, not working with events starting at the same time, etc) and that calendar triggers can support a lot of this functionality including matching events based on field names, etc.  

Would we ever consider holding off on adding all the UI configuration for calendar customization support here and work together on how we might support search for both google and caldav together?

Reply:
> Would we ever consider holding off on adding all the UI configuration for calendar customization support here and work together on how we might support search for both google and caldav together?

From my perspective, I'd be happy to adapt to however you want to design this as I don't have strong opinions here.

Reply:
I've hastily proposed something here to spur discussion https://github.com/home-assistant/architecture/discussions/776

Reply:
I think we should not add the custom calendars option to the config flow now.